src/hid_linux.c

```diff
@@ -268,11 +271,24 @@ fido_hid_open(const char *path)
 			fido_hid_close(ctx);
 			return (NULL);
 		}
-		interval_ms = retries * 100000000L;
-		tv_pause.tv_sec = interval_ms / 1000000000L;
-		tv_pause.tv_nsec = interval_ms % 1000000000L;
-		if (nanosleep(&tv_pause, NULL) == -1) {
-			fido_log_error(errno, "%s: nanosleep", __func__);
+		if (clock_gettime(CLOCK_MONOTONIC, &tv_pause) == -1) {
+			fido_log_error(errno, "%s: clock_gettime", __func__);
+			fido_hid_close(ctx);
+			return (NULL);
+		}
+		interval_ns = retries * 100000000L;
```

Comment:
Did you just rename the variable, because it is wrong in upstream or did you forget to also increase the multiplication constant?

Reply:
Maybe I misunderstood the previous name but I interpreted `interval_ms` as interval in milliseconds.

But since the conversion to seconds and nano seconds was done like this
```c
tv_pause.tv_sec = interval_ms / 1000000000L;
tv_pause.tv_nsec = interval_ms % 1000000000L;
```
I figured that its actually nano seconds. Otherwise it should have been:
```c
tv_pause.tv_sec = interval_ms / 1000000L;
tv_pause.tv_nsec = (interval_ms * 1000) % 1000000000L;
```

So what this `interval_ns = retries * 100000000L;` actually does is set the timeout to 0.1 seconds per number of retries.
|Retry|Timeout|
|--------|------------|
|1|0.1s|
|2|0.2s|
|...|...|
|10|1s|
|...|...|
|20|2s|